OpenCog installation

Good day,

Really sorry, for so maybe simple problem. I saw couple of your lectures and read some articles about OpenCog and wanted to try it, but I had some problems. I'm not sure that making issue is good idea, but I don't know where should I go with my questions.

I'm using Windows 7 build 7601. 
My VM Virtual box is version 5.1.12 r112440 (Qt5.6.2).
I've pulled master opencog 1 Jan, so it was commit 56666163bebd0eff7ac9341789ccba520b91060d.

I succesfully installed vagrant and up it from lib folder, because after "cd opencog" in instruction I can't find it in folder.

After vagrant ssh I try cmake opencog but occured error. Here is my terminal log:


E:\OpenCog\opencog\lib>vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'ubuntu/trusty32' is up to date...
==> default: A newer version of the box 'ubuntu/trusty32' is available! You curr
ently
==> default: have version '20160620.0.0'. The latest is version '20161214.0.0'.
Run
==> default: `vagrant box update` to update.
==> default: Setting the name of the VM: cogbox
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 5000 (guest) => 5000 (host) (adapter 1)
    default: 17001 (guest) => 17001 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Remote connection disconnect. Retrying...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version o
f
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you
 see
    default: shared folder errors, please make sure the guest additions within t
he
    default: virtual machine match the version of VirtualBox you have installed
on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 4.3.36
    default: VirtualBox Version: 5.0
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
==> default: Mounting shared folders...
    default: /vagrant => E:/OpenCog/opencog/lib
==> default: Machine already provisioned. Run `vagrant provision` or use the `--
provision`
==> default: flag to force provisioning. Provisioners marked to run always will
still run.

E:\OpenCog\opencog\lib>vagrant ssh
Welcome to Ubuntu 14.04.4 LTS (GNU/Linux 3.13.0-88-generic i686)

 * Documentation:  https://help.ubuntu.com/

  System information as of Mon Jan  2 10:46:50 UTC 2017

  System load:  0.68              Processes:           84
  Usage of /:   5.7% of 39.34GB   Users logged in:     0
  Memory usage: 3%                IP address for eth0: 10.0.2.15
  Swap usage:   0%                IP address for eth1: 192.168.50.2

  Graph this data and manage this system at:
    https://landscape.canonical.com/

  Get cloud support with Ubuntu Advantage Cloud Guest:
    http://www.ubuntu.com/business/services/cloud

New release '16.04.1 LTS' available.
Run 'do-release-upgrade' to upgrade to it.


Last login: Mon Jan  2 10:45:09 2017 from 10.0.2.2
vagrant@cogbox:~$ cd opencog
vagrant@cogbox:~/opencog$ mkdir build
vagrant@cogbox:~/opencog$ cd build
vagrant@cogbox:~/opencog/build$ cmake ..
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Error: File /home/vagrant/opencog/lib/opencog-chatbot.conf does not exist.

CMake Error at CMakeLists.txt:1 (CONFIGURE_FILE):
  configure_file Problem configuring file


CMake Error: File /home/vagrant/opencog/lib/opencog.conf does not exist.
CMake Error at CMakeLists.txt:3 (CONFIGURE_FILE):
  configure_file Problem configuring file


CMake Error: File /home/vagrant/opencog/lib/opencog-test.conf does not exist.
CMake Error at CMakeLists.txt:6 (CONFIGURE_FILE):
  configure_file Problem configuring file


CMake Error at CMakeLists.txt:19 (INSTALL):
  install FILES given no DESTINATION!


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 2.8)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
See also "/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log".

In CMakeOutput.log was next logs:

vagrant@cogbox: /opencog/build$ /home/vagrant/opencog/build/CMakeFiles/CMakeOut
put.log
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 1: The: command not
 found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 2: Compiling: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 3: Compiler:: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 4: Build: command n
ot found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 5: Id: command not
found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 7: The: command not
 found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 8: 0: command not f
ound
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 11: Compilation: co
mmand not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 13: The: command no
t found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 15: Compiling: comm
and not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 16: Compiler:: comm
and not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 17: Build: command
not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 18: Id: command not
 found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 20: The: command no
t found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 21: 0: command not
found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 24: Compilation: co
mmand not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 26: The: command no
t found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 28: Determining: co
mmand not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 29: Change: command
 not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 31: Run: command no
t found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 32: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 35: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 32: make:: command
not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 37: Building: comma
nd not found
cc: error: /home/vagrant/opencog/build/CMakeFiles/CMakeTmp/testCCompiler.c: No s
uch file or directory
cc: fatal error: no input files
compilation terminated.
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 39: Linking: comman
d not found
Error opening link script "CMakeFiles/cmTryCompileExec2048697173.dir/link.txt"
cc: error: CMakeFiles/cmTryCompileExec2048697173.dir/testCCompiler.c.o: No such
file or directory
cc: fatal error: no input files
compilation terminated.
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 42: make[1]:: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 43: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 52: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 43: make[1]:: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 70: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 74: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 54: make[1]:: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 83: ignore: command
 not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 84: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 86: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 84: ignore: command
 not found
make: CMakeFiles/cmTryCompileExec29222649.dir/build.make: No such file or direct
ory
make: *** No rule to make target `CMakeFiles/cmTryCompileExec29222649.dir/build.
make'.  Stop.
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 163: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 165: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 163: make[1]:: comm
and not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 173: make:: command
 not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 176: Detecting: com
mand not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 177: Change: comman
d not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 179: Run: command n
ot found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 180: unexpected EOF while looking for matching `''
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: command substitution: li
ne 183: syntax error: unexpected end of file
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 180: make:: command
 not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 185: Building: comm
and not found
Assembler messages:
Fatal error: can't create CMakeFiles/cmTryCompileExec2479775140.dir/CMakeCXXComp
ilerABI.cpp.o: No such file or directory
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 187: Linking: comma
nd not found
Error opening link script "CMakeFiles/cmTryCompileExec2479775140.dir/link.txt"
c++: error: CMakeFiles/cmTryCompileExec2479775140.dir/CMakeCXXCompilerABI.cpp.o:
 No such file or directory
Using built-in specs.
COLLECT_GCC=/usr/bin/c++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-linux-gnu/4.8/lto-wrapper
Target: i686-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.8.4-2ubuntu1~14
.04.3' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-language
s=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --ena
ble-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-get
text --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir
=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-
debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libmudfla
p --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=
gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-i386/jre
 --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-i386 --w
ith-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-i386 --with-arch-directo
ry=i386 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable
-targets=all --enable-multiarch --disable-werror --with-arch-32=i686 --with-mult
ilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=i68
6-linux-gnu --host=i686-linux-gnu --target=i686-linux-gnu
Thread model: posix
gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3)
c++: fatal error: no input files
compilation terminated.
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 190: Using: command
 not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 193: Target:: comma
nd not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 194: Configured: co
mmand not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 195: Thread: comman
d not found
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 196: syntax error n
ear unexpected token `('
/home/vagrant/opencog/build/CMakeFiles/CMakeOutput.log: line 196: `gcc version 4
.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04.3) '

Will be appreciated for any suggestions.

--- a/lib/Vagrantfile
+++ b/lib/Vagrantfile
@@ -29,7 +29,6 @@ Vagrant.configure("2") do |config|
   # 64-bit machine caused a fatal issue: http://stackoverflow.com/a/22575302/1695962
   config.vm.box = "ubuntu/trusty32"
   config.vm.hostname = "cogbox"
-  config.vm.provision "shell", inline: "ln -v -s /vagrant /home/vagrant/opencog"
   config.vm.provision "shell", inline: "wget https://raw.githubusercontent.com/opencog/ocpkg/master/ocpkg -O /tmp/octool  && chmod +rx /tmp/octool"
   config.vm.provision "shell", inline: "/tmp/octool -rdpscalv"
 
@@ -42,6 +41,9 @@ Vagrant.configure("2") do |config|
 
   # Configure port for telnet access to shell
   config.vm.network "forwarded_port", guest: 17001, host: 17001
+
+  # Sync project root directory.
+  config.vm.synced_folder "../", "/home/vagrant/opencog"
 
   # Enable X-Forwarding
   config.ssh.forward_x11 = true

